torch/accelerator/__init__.py

```diff
@@ -0,0 +1,140 @@
+# mypy: allow-untyped-defs
+r"""
+This package introduces support for the current :ref:`accelerator<accelerators>` in python.
+"""
+
+from typing import Any
+
+import torch
+
+from ._utils import _device_t, _get_device_index
+
+
+def current_accelerator() -> torch.device:
+    r"""Return the device of the current :ref:`accelerator<accelerators>`. And use
+    :func:`torch.accelerator.is_available` to check if it is available.
+    Note that the index of the returned :class:`torch.device` will be ``None``, use
+    :func:`torch.accelerator.current_device_idx` to know the current index being used.
+    This function doesn't gaurantee that the current accelerator is available, use
+    :func:`torch.accelerator.is_available` to check its availability.
+
+    Returns:
+        torch.device: return the current accelerator as :class:`torch.device`.
+            If no available accelerators, return cpu device.
```

Comment:
I have a question. Is it good to return CPU if no available accelerator?
Pros: return CPU will facilitate the code usage, for example below. We can use one line code to create a tensor in current accelerator or CPU. It is benefit to unify the CPU and accelerator code if necessary.
```python
tensor = torch.randn(100, device=torch.accelerator.current_accelerator())
```
Cons:
1. I really believe we will need a API named `torch.accelerator.is_accelerator()` in the future. At that time, the below code looks weird. `torch.accelerator.is_accelerator(torch.accelerator.current_accelerator())` possibly could return `False`. It means **current accelerator is not a accelerator**. The logic is not self-consistent.

Reply:
Or we design like this
```python
def current_accelerator(fallback=False): ->Optional(Device)
    if accelerator is available:
        return accelerator
    return CPU if fallback else None
```

Reply:
I do think we shouldn't have cpu here. And should be in the full device API as discussed below.

Reply:
OK. change to rase an error if no available accelerator is detected.